### DISCUSSION TITLE
Problem with spawning new NPCs

### DISCUSSION THREAD

**Discussion by Cervicia** [MattFiler/LegendPlugin#193] [Q&A] [answered]
Hello,
I tried to spawn new NPCs, specifically humans in the TECH_HUB level, but they won't spawn neither through cloning existing ones nor through creating new ones. The only thing that spawns is the model of their weapon. Spawning Androids, Aliens and moving the spawns of existing humans is no problem. Do you have any idea, what might be the problem?
<img width="952" alt="AI_NPC_weapon" src="https://user-images.githubusercontent.com/100235168/236488988-eae4089a-6a7b-482a-b129-a900c997ea6f.PNG">

**Comment by MattFiler** [comment, marked as answer]
Thanks for the report! If you're completely copying the setup of other NPCs then it may be that they are somehow handled elsewhere that the scripting tool doesn't talk to in the game yet. Could you create an issue with this info so I can keep a track of it?

Cheers!

**Comment by ghramsey** [comment]
I've been a late comer to using the script editor; only since November last year.

I have notes from January where I moved the corpse trap from the end of M4 to appear in M3.  
At the same time I was also trying to move the humans who spawn at on the tram station during that portion 
of the game and I too noticed I could not make them appear even by reproducing the script for them in the 
M3 mission script.  (I wanted some alien fodder mwhahaha).

At the time I thought it was just something I was doing wrong. I'll be interested to find out how this is solved.
Being able to spawn at will, or otherwise manipulate the NPCs will add some interesting aspects to the game.

**Reply by MattFiler** [reply]
It turned out human NPCs are setup using a whole other file. I've written support for it into the script editor now and it's working - still got a few more things to figure out and then I'll push it to staging. Here's a pic of an NPC spawned from scratch: https://github.com/MattFiler/OpenCAGE/issues/194#issuecomment-1567550201

**Comment by ghramsey** [comment]
OOOOOHHH THAT's cool!  THank you for all your hard work!  *imagines the possibilities*

**Reply by MattFiler** [reply]
First version of the editor is now live on staging!

Here's some steps to get going: https://github.com/MattFiler/OpenCAGE/issues/194#issuecomment-1569166976
